### ISSUE TITLE
[v6] Error generating from latest schema registry swagger: duplicate object schemas

### ISSUE THREAD

**Issue by nguerrera** [Azure/autorest.typescript#736]
I'm trying to update schema-registry generated code for TypeScript

Given this configuration:
``` yaml
description: Generated Schema Registry Client
generate-metadata: false
add-credentials: false
license-header: MICROSOFT_MIT_NO_VERSION
output-folder: ../
source-code-folder-path: ./src/generated
input-file: https://raw.githubusercontent.com/Azure/azure-rest-api-specs/1e23d91e875e4464e57667639e06408cef99868d/specification/schemaregistry/data-plane/Microsoft.EventHub/preview/2020-09-01-preview/schemaregistry.json
use-extension:
  "@autorest/typescript": "6.0.0-dev.20200918.1"
```

I get:
```
ERROR (): Duplicate object schemas with 'SerializationType' name  detected.
FATAL: Error: 1 errors occured -- cannot continue.
  Error: Plugin checker reported failure.
```

I am not sure if this is an issue with the swagger authoring (cc @arerlend)  or with the generator (cc @joheredi).

**Comment by joheredi** [comment]
Hi @nguerrera, thanks for reporting. I'll take a look and get back to you when I find more info

**Comment by nguerrera** [comment]
Thanks. I was able to work around this with a transform that removed the enums I don’t need so I’m not blocked.

**Comment by sarangan12** [comment]
I have checked this file. The same error is reported. But, it does not happen within the TS code. It actually happens within the core autorest code even before ```generateTypeScriptLibrary``` method is executed. So, I will loop in @daviwil to take a look at this issue.

**Comment by MiYanni** [comment]
I have details on this issue and discussed with David. I will be creating an AutoRest issue for it. I'll link back to this one.

**Comment by daviwil** [comment]
Yeah, this one is tricky because there's duplication of the `x-ms-enum` everywhere but used in different schema types so it's not easily deduplicatable.  Due to the way the schema is designed, it's hard to reuse the `x-ms-enum` definition so I'll have to find a way to deduplicate it in the modeler phase.

**Comment by MiYanni** [comment]
Swagger doesn't allow you to ref schemas in headers... so deduplication in AutoRest is the only option here. @daviwil The intent is to use the same schema for both request header parameters and response headers. So, the definition of those enums should be identical. I had checked that they were when I ran into this issue for C#. So, header handling (both as parameters and responses) might need to be done differently to account for this situation.

**Comment by MiYanni** [comment]
Created an issue for it here: https://github.com/Azure/autorest.modelerfour/issues/362

**Comment by sarangan12** [comment]
@nguerrera As discussed in the above conversations, the issue is happening with modelerfour and it would be best answered if you follow it up with the issue https://github.com/Azure/autorest/issues/3629. 

If you still face any issue after the modeler issue is fixed, feel free to reopen the issue.

No more pending action items for this issue. Closing it now.

**Comment by MiYanni** [comment]
@sarangan12 I've reopened the issue you mentioned so it can be investigated. It was improperly tested in the issue. You can keep this issue closed.
